test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionEnforcementTest.java

```diff
@@ -40,6 +48,103 @@
 
 public final class InternodeEncryptionEnforcementTest extends TestBaseImpl
 {
+
+    @Test
+    public void testConnectionsAreRejectedWhenAuthFails() throws IOException, InterruptedException
+    {
+        Cluster.Builder builder = builder()
+        .withNodes(2)
+        .withConfig(c ->
+                    {
+                        c.with(Feature.NETWORK);
+                        c.with(Feature.NATIVE_PROTOCOL);
+
+                        HashMap<String, Object> encryption = new HashMap<>();
+                        encryption.put("keystore", "test/conf/cassandra_ssl_test.keystore");
+                        encryption.put("keystore_password", "cassandra");
+                        encryption.put("truststore", "test/conf/cassandra_ssl_test.truststore");
+                        encryption.put("truststore_password", "cassandra");
+                        encryption.put("internode_encryption", "dc");
+                        encryption.put("require_client_auth", "true");
+                        c.set("server_encryption_options", encryption);
+                        c.set("internode_authenticator", AllowNothingAuthenticator.class.getName());
+                    })
+        .withNodeIdTopology(ImmutableMap.of(1, NetworkTopology.dcAndRack("dc1", "r1a"),
+                                            2, NetworkTopology.dcAndRack("dc2", "r2a")));
+
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        try (Cluster cluster = builder.start())
+        {
+            executorService.submit(() -> {
+                openConnections(cluster);
+            });
+
+            /*
+             * instance (1) should connect to instance (2) without any issues;
+             * instance (2) should connect to instance (1) without any issues.
+             */
```

Comment:
Comment does not match code.

How do you know the failed connection was due to the authenticator failing the attempt rather than another issue? You could add an @Shared class that either holds counters or a log of authentication attempts in the test.

Reply:
Updated the test to verify that.